src/lib.rs

```diff
@@ -1,3 +1,20 @@
+//! An experimental implementation of [Rust RFC#2256 libsyntax2.0][rfc#2256].
+//!
+//! The intent is to be an IDE-ready parser, i.e. one that offers
+//!
+//! - easy and fast incremental re-parsing,
+//! - graceful handling of errors, and
+//! - maintains all information in the source file.
+//!
+//! For more information, see [the RFC][rfc#2265], or [the working draft][RFC.md].
+//!
+//!   [rfc#2256]: <https://github.com/rust-lang/rfcs/pull/2256>
+//!   [RFC.md]: <https://github.com/matklad/libsyntax2/blob/master/docs/RFC.md>
+
+#![forbid(missing_debug_implementations, unconditional_recursion, future_incompatible)]
+#![deny(bad_style, unsafe_code, missing_docs)]
```

Comment:
Hm, I feel slightly uneasy about `missing_docs`: this is very experimental and will change a lot, so all docs potentially will have to be rewritten many times.

On the other hand, I hope that the public API surface will be small, and documenting it would help with design...

This also exposed one interesting problem: `FileBuilder` is not a public API and should be `pub(crate)`.